### ISSUE TITLE
rmmod: ERROR: Module bus1 is in use

### ISSUE THREAD

**Issue by EduRam** [bus1/bus1#18]
I cannot unload bus1 module.
It complains about still being used.
This happens after running bus1 own tests 
(but with error)

I don't know how to identify who is using it !!!
Please fell free to discard this issue if there is work in progress ...

```
[root@dock bus1]# rmmod bus1
rmmod: ERROR: Module bus1 is in use

[root@dock bus1]# lsmod | grep bus1
bus1                   53248  2

[root@dock bus1]# modinfo bus1
filename:       /lib/modules/4.8.0-0.rc3.git0.1.fc26.x86_64/kernel/ipc/bus1/bus1.ko
description:    Bus based interprocess communication
license:        GPL
depends:        
vermagic:       4.8.0-0.rc3.git0.1.fc26.x86_64 SMP mod_unload 
parm:           user_slices_max:Max number of slices for each user. (ushort)
parm:           user_handles_max:Max number of handles for each user. (ushort)
parm:           user_bytes_max:Max number of bytes for each user. (uint)
parm:           user_fds_max:Max number of fds for each user. (ushort)
```

**Comment by dvdhrm** [comment]
Can you tell us the error?

**Comment by EduRam** [comment]
Hi! I add a note of the test error on Issue #2 (see last comment).

**Comment by dvdhrm** [comment]
Your error really looks like a kernel-header mismatch between your running kernel and the installed headers. I just verified that everything is fine with 4.8-rc3.

Are you sure that you have the headers for 4.8-rc3 installed? That is, do the headers match the kernel version you actually run? (see `uname -r`)

**Comment by EduRam** [comment]
Hi! 

I reinstalled a fresh F24 guest on a HyperV host.
I used this time the current F24 debug kernel (4.6.7-300.fc24.x86_64+debug).
The attachement have the kernel log.
This time there are some extra debug lines that may help.

> BUG: sleeping function called from invalid context at include/linux/sched.h:2814
> in_atomic(): 0, irqs_disabled(): 1, pid: 1470, name: bus1-test

[bus1tests_on_vmguest.txt](https://github.com/bus1/bus1/files/439829/bus1tests_on_vmguest.txt)

**Comment by dvdhrm** [comment]
Should be fixed in e71a560335adfc30f4d88cc35ca98afbe55d369f.

**Comment by teg** [comment]
Please reopen if this is still an issue.
